src/fluidplayer.js

```diff
@@ -1910,41 +1911,48 @@ const fluidPlayerClass = function () {
         };
         let initiateVolumebarTimerId = setInterval(initiateVolumebar, 100);
         self.destructors.push(() => clearInterval(initiateVolumebarTimerId));
-
-        if (self.displayOptions.layoutControls.doubleclickFullscreen && !(self.isTouchDevice() || self.displayOptions.layoutControls.controlForwardBackward.doubleTapMobile)) {
-            self.domRef.player.addEventListener('dblclick', self.fullscreenToggle);
+    
+        // Separate behavior for desktop and mobile
+        if (self.displayOptions.layoutControls.doubleclickFullscreen && !self.isTouchDevice()) {
+            self.domRef.player.addEventListener('dblclick', () => {
+                console.log('Double click detected for fullscreen'); // Optional for debugging
+                self.fullscreenToggle();
+            });
+        } else if (self.isTouchDevice() && !!self.displayOptions.layoutControls.controlForwardBackward.doubleTapMobile) {
+            self.initDoubleTapSkip();
```

Comment:
Because you initialise the doubleTapSkip here,  it' initialised twice. So now it's skipping 20 seconds